django_socio_grpc/services/servicer_proxy.py

```diff
@@ -264,32 +271,40 @@ def __getattr__(self, action):
 
     def process_exception(self, exc, request_container: GRPCRequestContainer):
         if isinstance(exc, GRPCException):
-            logger.error(
-                f"{type(exc).__name__} : {exc}",
-                exc_info=exc,
-            )
             request_container.context.abort(exc.status_code, exc.get_full_details())
-        elif isinstance(exc, grpc.RpcError) or request_container.context._state.aborted:
-            raise exc
         else:
-            logger.error(
-                f"{type(exc).__name__} : {exc}",
-                exc_info=exc,
-            )
             request_container.context.abort(grpc.StatusCode.UNKNOWN, str(exc))
 
-    async def async_process_exception(self, exc, request_container: GRPCRequestContainer):
+    async def async_process_exception(self, exc, context):
         if isinstance(exc, GRPCException):
-            logger.error(
-                f"{type(exc).__name__} : {exc}",
-                exc_info=exc,
-            )
-            await request_container.context.abort(exc.status_code, exc.get_full_details())
-        elif isinstance(exc, (grpc.RpcError, grpc.aio.AbortError)):
-            raise exc
+            await context.abort(exc.status_code, exc.get_full_details())
+        else:
+            await context.abort(grpc.StatusCode.UNKNOWN, str(exc))
+
+    def log_request_success(self, response, request_container):
+        request_logger.info(
+            f"{type(request_container.service).__name__}.{request_container.service.action} : {response.http_response.status_code}"
+        )
+
+    def log_exception(self, exception, request_container):
+        if isinstance(exception, GRPCException):
+            if exception.logging_level == "INFO":
+                request_logger.info(
+                    f"{type(request_container.service).__name__}.{request_container.service.action} : {exception.status_code.name}",
+                    exc_info=exception,
+                )
+            if exception.logging_level == "WARNING":
+                request_logger.warning(
+                    f"{type(request_container.service).__name__}.{request_container.service.action} : {exception.status_code.name}",
+                    exc_info=exception,
+                )
+            if exception.logging_level == "ERROR":
+                request_logger.error(
+                    f"{type(request_container.service).__name__}.{request_container.service.action} : {exception.status_code.name}",
+                    exc_info=exception,
+                )
```

Comment:
simplify this

Reply:
any idea?

Reply:
The string is the same for all, and the logging level has the name of the function called on the logger